crates/typst-syntax/src/lexer.rs

```diff
@@ -343,7 +344,15 @@ impl Lexer<'_> {
     }
 
     fn in_word(&self) -> bool {
-        let alphanum = |c: Option<char>| c.map_or(false, |c| c.is_alphanumeric());
+        let alphanum = |c: Option<char>| {
```

Comment:
The name `alphanum` is a bit misleading here now, right? It should probably be changed.

Reply:
What do you think is a good name? Do you like the name `non_cjk_alphanum`?

Reply:
Either something descriptive like that or something like `wordy` in line with `in_word`.